/* HERE IN THIS PART , WE WILL DISCUSS ABOUT THE EXCEPTIONAL HANDLING FEATURE AVAILABLE IN  C++
EXCEPTIONAL HANDLING IS A FEATURE OF C++ WHICH HANDLES THE UNUSAL CONDITIONS DURING RUN TIME OR 
AT THE TIME OF EXECUTION....

THE WHOLE EXCEPTIONAL HANDLING IN C++ IS BASED ONLY UPON THREE KEYWORDS TRY,CATCH AND THROW..
EXCPTIONAL HANDLING HAS TWO SEGMENTS OF ERROR HANDLING CODE , 1ST SEGEMENT DETECTS AN 
EXCEPTION AND THROWS IT WHILE OTHER CATCHES IT AND TAKE NECESSARY ACTION AND DISPLAY 
APPROPRIATE MESSAGE TO THE USER....*/

#include<iostream>
using namespace std;

int main()
{
    //BASIC TRY-CATCH BLOCK EXCEPTIONAL HANDLING CODE.....
    int x,y;
    cout<<"ENTER 1st AND 2nd VALUE : "<<endl;
    cin>>x>>y;
    try
    {
        cout<<"INSIDE THE TRY BLOCK....\n";
        if(y==0)
        {
            throw y;
        }
        else
        {
            cout<<"EXCEPTION IS NOT CAUGHT...";
        }
    }
    //REMEMBER THAT CATCH BLOCK MUST BE THERE JUST AFTER TRY BLOCK...
    catch(float y) //HANDLER , IF THROW OBJECT EXCEPTION TYPE DOESN'T MATCH WITH THE CATCH ARGUMENT THEN PROGRAM WILL TERMINATE ABNORMALLY....
    {
        cout<<"DIVISION BY 0 EXCEPTION ARRISES....\n";
    }
    cout<<"END OF MAIN....";
    return 0;
}